cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -467,6 +496,20 @@ func isDeviceStatusUpdated(oldTwin *v1alpha2.DeviceStatus, newTwin *v1alpha2.Dev
 	return !reflect.DeepEqual(oldTwin, newTwin)
 }
 
+// isDesiredTwinUpdated checks if desired twin is updated
+func isDesiredTwinUpdated(oldTwinDesired *v1alpha2.DeviceStatus, newTwin *v1alpha2.DeviceStatus) bool {
+	index := 0
+	res := true
+	for index < len(oldTwinDesired.Twins) && index < len(newTwin.Twins) {
+		res = res && reflect.DeepEqual(oldTwinDesired.Twins[index].Desired, newTwin.Twins[index].Desired)
+		index++
+	}
+	if index != len(oldTwinDesired.Twins) || index != len(newTwin.Twins) {
+		return true
+	}
```

Comment:
use `if len(oldTwinDesired.Twins) != len(newTwin.Twins)` looks better, and it can be write at first

Reply:
Excellent advice. I will write at first.